core/sodasql/soda_server_client/soda_server_client.py

```diff
@@ -42,41 +42,75 @@ def __init__(self,
         self.api_key_secret: Optional[str] = api_key_secret
         self.token: Optional[str] = token
 
-    def scan_start(self, warehouse, scan_yml: ScanYml, scan_time,
-                   origin: str = 'external'):
+    def scan_start(
+        self,
+        warehouse_name: str,
+        warehouse_type: str,
+        warehouse_database_name: Optional[str],
+        warehouse_database_schema: Optional[str],
+        table_name: str,
+        scan_yml_columns: Optional[Dict[str, ScanYmlColumn]],
+        scan_time: str,
+        origin: str = "external",
+    ):
+        """
+        Start the scan.
+
+        Parameters
+        ----------
+        warehouse_name : str
+            The warehouse name.
+        warehouse_type : str
+            The warehouse (and dialect) type.
+        warehouse_database_name : Optional[str]
```

Comment:
I moved these to the arguments, so that they do not require the `warehouse` object